torch/_inductor/codegen/common.py

```diff
@@ -1030,8 +1029,13 @@ def indirect_indexing(var, size, check=True):
 
                     new_var.update_on_args("index_wrap", (var,), {})
                     var = new_var
+                return var
+
+            @staticmethod
+            def check_bounds(var, size):
+                var = CSEProxy._indirect_to_direct_index(var, size)
```

Comment:
`var` shouldn't be an indirect index here

Reply:
Although, this does make me think. A rather hacky way to do this might be
```python
ops.indirect_indexing(ops.index_expr(var, torch.int32), size, check=True)
```